src/controller.cc

```diff
@@ -174,12 +173,15 @@ bool Controller::AddTransaction(Transaction trans) {
         }
         trans.complete_cycle = clk_ + 1;
         return_queue_.push_back(trans);
+        last_trans_clk_ = clk_;
         return true;
     } else {  // read
+        simple_stats_.AddValue("interarrival_read_latency", clk_ - trans.start_cycle);
```

Comment:
This is the time spent in the read queue (`read_queue_`). I would call it that.

In fact, I would add a "time in read queue" and a "time in write queue" stat.

PS: Another thing related to what we were talking about last night. Communication is very important. Research is all about communicating what you discover to others. This is one reason why variable/stat names are very important. Make sure you name things *what you want others* to take away. Don't use names that only you (or the author of the code) understand.